include/grpcpp/impl/codegen/callback_common.h

```diff
@@ -182,11 +183,11 @@ class CallbackWithSuccessTag
   }
   void Run(bool ok) {
     void* ignored = ops_;
-    bool new_ok = ok;
     // Allow a "false" return value from FinalizeResult to silence the
     // callback, just as it silences a CQ tag in the async cases
-    bool do_callback = ops_->FinalizeResult(&ignored, &new_ok);
-    GPR_CODEGEN_ASSERT(ignored == ops_);
+    auto* ops = ops_;
+    bool do_callback = ops_->FinalizeResult(&ignored, &ok);
+    GPR_CODEGEN_ASSERT(ignored == ops);
 
     if (do_callback) {
       CatchingCallback(func_, ok);
```

Comment:
We should probably catch func_ before calling FinalizeResult as well

Reply:
I know that I had suggested this as a possible area of improvement when we discussed it, but I'd now like to push back if possible. A function isn't trivially copyable like a pointer (like `ops`), so I'd prefer not to grab it unless it's actually needed. The call to `CatchingCallback` probably gets inlined so there's no copy there (since the function argument is taken as an rvalue reference). Right now, there is no circumstance where `do_callback` will be true but the `FinalizeResult` function releases the arena. This only happens in situations where `FinalizeResult` ends up returning false. If this decision changes and we eventually do return true while unreffing the arena, ASAN will warn us of the problem (believe me, ASAN issues were the main reason these PRs took so many extra days from initial send to "reviewable"). Putting all of this together, I'd like to avoid grabbing `func_` before this call.